### ISSUE TITLE
SENTRY_OPTIONS['mail.use-tls'] is ignored

### ISSUE THREAD

**Issue by mastacheata** [getsentry/sentry#3077]
Sentry on-premise v8.3.1 here.
Whatever I enter in config.yml for mail.use-tls, sentry will ignore it and revert to the default of false.
I tried:
```
mail.use-tls: true
mail.use-tls: 'true'
```

Neither seems to work: https://dl.dropboxusercontent.com/u/7370504/sentry-mail.use-tls.png
Sentry will always tell me it's found tls to be off.
Which in turn leads to error messages like ths:
`SMTPException: SMTP AUTH extension not supported by server.`
Because my mail server prohibits logins via insecure plain SMTP.

My workaround for the time being: Keep setting the EMAIL_USE_TLS option in sentry.conf.py and ignore the deprecation message.

**Comment by mattrobenolt** [comment]
Hmm, lemme investigate this real quick. I swear this was working.

**Comment by mattrobenolt** [comment]
Yeah, I even have tests that cover this: https://github.com/getsentry/sentry/blob/master/tests/sentry/runner/test_initializer.py

@mastacheata do you happen to have both `EMAIL_USE_TLS` and `mail.use-tls` set?

**Comment by mattrobenolt** [comment]
Also, do you do something weird with email backends? Do you use a custom email backend, etc? Because the built in mail backends shouldn't even be sourcing credentials from `EMAIL_USE_TLS`. It's explicitly being sourced out of the new stuff: https://github.com/getsentry/sentry/blob/master/src/sentry/utils/email.py#L372-L384

**Comment by mastacheata** [comment]
Nope. When I set the config.yml one, I uncomment the other one.

sentry.conf.py
```
###############
# Mail Server #
###############

# For more information check Django's documentation:
# https://docs.djangoproject.com/en/1.6/topics/email/

#EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'

#EMAIL_HOST = 'localhost'
#EMAIL_HOST_PASSWORD = 'xxxxxxx'
#EMAIL_HOST_USER = 'sentry@ymoq.de'
#EMAIL_PORT = 25
#EMAIL_USE_TLS = True

# The email address to send on behalf of
#SERVER_EMAIL = 'sentry@ymoq.de'
```
config.yml
```
mail.backend: 'smtp'
mail.host: 'localhost'
mail.password: 'xxxxxxxxxx'
mail.user: 'sentry@ymoq.de'
mail.port: '25'
mail.use-tls: true
mail.from: 'sentry@ymoq.de'
```


Don't think this is any weird config at all, see screenshot in original issue for what sentry reports in the web backend.

**Comment by mattrobenolt** [comment]
I just played with this locally, and I can't seem to reproduce. When you get an error, can you paste the full stacktrace?

**Comment by mastacheata** [comment]
Sure. This is what I get when defining use-tls only in config.yml:
```
SMTPException: SMTP AUTH extension not supported by server.
  File "celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "sentry/tasks/base.py", line 54, in _wrapped
    result = func(*args, **kwargs)
  File "sentry/tasks/digests.py", line 71, in deliver_digest
    plugin.notify_digest(project, digest)
  File "sentry/plugins/sentry_mail/models.py", line 232, in notify_digest
    message.send()
  File "sentry/utils/email.py", line 348, in send
    fail_silently=fail_silently,
  File "sentry/utils/email.py", line 361, in send_messages
    return connection.send_messages(messages)
  File "django/core/mail/backends/smtp.py", line 87, in send_messages
    new_conn_created = self.open()
  File "django/core/mail/backends/smtp.py", line 54, in open
    self.connection.login(self.username, self.password)
  File "python2.7/smtplib.py", line 585, in login
    raise SMTPException("SMTP AUTH extension not supported by server.")
```

Note that "SMTP AUTH extension not supported by server" is happening when you omit the starttls() call on the smtp connection in python.

**Comment by mastacheata** [comment]
The problem only shows if your smtp server enforces secure logins. (Take GMail for example)
If the SMTP Server accepts login without STARTTLS, you will never see this error.

**Comment by mattrobenolt** [comment]
The only thing I can think of is that your celery workers are running against a different config, or haven't been restarted. I can't see any reason this would fail. You can double check by running a few sanity checks:

```
$ sentry shell
...
from django.conf import settings
print(settings.EMAIL_USE_TLS)
print(settings.SENTRY_OPTIONS)
from sentry import options
print(options.get('mail.use-tls'))
```

```
$ sentry config get mail.use-tls
```

But I just can't see how this is possible since the `EMAIL_USE_TLS` setting shouldn't even be used in that stacktrace.

**Comment by mastacheata** [comment]
Hmm. Removed config files and started fresh.
That seems to have helped.

Apparently my config.yml was wrong all the time.
The port number is a string => Seems python doesn't like that.
mail.user is an unknown config option, it must be mail.username

These errors were fatal errors for sentry upgrade, but weren't even mentioned in sentry start command.
They weren't logged to sentry itself either.

It looks like sentry either didn't parse the rest of the file if these options were bad.

I did an upgrade install and hadn't created a new config.yml with out-commented email config, but typed these by hand according to your guides.
Apparently I just skimmed through that documentation and simply used lowercase versions of the sentry.conf.py variables/constants/settings and didn't notice the username vs user change.

Sorry, this seems to be my fault then.
